source/extensions/filters/network/redis_proxy/proxy_filter.cc

```diff
@@ -16,10 +17,14 @@ namespace RedisProxy {
 
 ProxyFilterConfig::ProxyFilterConfig(
     const envoy::config::filter::network::redis_proxy::v2::RedisProxy& config, Stats::Scope& scope,
-    const Network::DrainDecision& drain_decision, Runtime::Loader& runtime)
+    const Network::DrainDecision& drain_decision, Runtime::Loader& runtime, Api::Api& api)
     : drain_decision_(drain_decision), runtime_(runtime),
       stat_prefix_(fmt::format("redis.{}.", config.stat_prefix())),
-      stats_(generateStats(stat_prefix_, scope)) {}
+      stats_(generateStats(stat_prefix_, scope)),
+      downstream_auth_password_(
+          Config::DataSource::read(config.downstream_auth_password(), true, api)) {
+  auth_required_ = !downstream_auth_password_.empty();
```

Comment:
Nit: This isn't used much- easier to just check for `connection_allowed_ = config-downstream_auth_password.empty()`?

Reply:
yeah, I missed that -- thanks. It was originally going to be more complicated...sigh...